src/engines/html5/html5.js

```diff
@@ -941,4 +947,57 @@ export default class Html5 extends FakeEventTarget implements IEngine {
     }
     return extended;
   }
+
+  enterPictureInPicture(): void {
+    try {
+      // Currently it's supported in chrome and in safari. So if we consider checking support before,
+      // we can use this flag to distinguish between two. In the future we might need a different method.
+      if (document.pictureInPictureEnabled) {
+        this._el
+          .requestPictureInPicture()
+          .catch(error =>
+            this.dispatchEvent(new Error(Error.Severity.RECOVERABLE, Error.Category.PLAYER, Error.Code.CANNOT_ENTER_PICTURE_IN_PICTURE, error))
+          );
+      } else {
+        this._el.webkitSetPresentationMode('picture-in-picture');
+      }
+      this._isInPictureInPicture = true;
+    } catch (error) {
+      this.dispatchEvent(new Error(Error.Severity.RECOVERABLE, Error.Category.PLAYER, Error.Code.CANNOT_ENTER_PICTURE_IN_PICTURE, error));
+    }
+  }
+
+  exitPictureInPicture(): void {
+    try {
+      // Currently it's supported in chrome and in safari. So if we consider checking support before,
+      // we can use this flag to distinguish between two. In the future we might need a different method.
+      if (document.pictureInPictureEnabled) {
+        document
+          .exitPictureInPicture()
+          .catch(error =>
+            this.dispatchEvent(new Error(Error.Severity.RECOVERABLE, Error.Category.PLAYER, Error.Code.CANNOT_EXIT_PICTURE_IN_PICTURE, error))
+          );
+      } else {
+        this._el.webkitSetPresentationMode('inline');
```

Comment:
in safari emit the player leave pip event ourselves, possibly make it async(if this is what chrome does) using setTimeout 0